### PR TITLE
Add tests for h.directory()

### PR DESCRIPTION
**Pull request by venikman**

### CODE REVIEW

File: test/toolkit.js [hunk 1]
```diff
@@ -124,3 +124,81 @@ test('h.file() outside default confine', async () => {
...
+    assertStrictEq(response.status, 200);
+    assertStrictEq(response.headers.get('content-type'), 'application/json; charset=utf-8');
+    assertEquals(response.body, new TextEncoder().encode('[\n    "Alice",\n    "Bob",\n    "Cara"\n]\n'));
+});
+
+test('h.directory() listing with relative path', async () => {
+    const server = pogo.server();
+    server.route({
+        method : 'GET',
+        path   : '/dirtest/{dirname?}',
```
Comment (sholladay) [inline test/toolkit.js:150]: Let's either rename the parameter to `{file*}` to match the first test, or just remove the parameter since we don't use it when making the request.

File: test/toolkit.js [hunk 2]
```diff
@@ -124,3 +124,81 @@ test('h.file() outside default confine', async () => {
...
+    assertStrictEq(response.headers.get('content-type'), 'text/html; charset=utf-8');
+    assertStringContains(response.body, '<html>');
+    assertStringContains(response.body, 'names.json');
+});
+
+test('h.directory() listing with partial relative path', async () => {
+    const server = pogo.server();
+    server.route({
+        method : 'GET',
+        path   : '/dirtest/{dirname?}',
```
Comment (sholladay) [inline test/toolkit.js:169]: Same here, either rename the parameter to `{file*}` to match the first test, or just remove the parameter since we don't use it when making the request.

File: test/toolkit.js [hunk 3]
```diff
@@ -124,3 +124,81 @@ test('h.file() outside default confine', async () => {
...
+    assertStrictEq(response.headers.get('content-type'), 'text/html; charset=utf-8');
+    assertStringContains(response.body, '<html>');
+    assertStringContains(response.body, 'names.json');
+});
+
+test('h.directory() forbids listing by default', async () => {
+    const server = pogo.server();
+    server.route({
+        method : 'GET',
+        path   : '/dirtest/{dirname?}',
```
Comment (sholladay) [inline test/toolkit.js:188]: Same here, either rename the parameter to `{file*}` to match the first test, or just remove the parameter since we don't use it when making the request.

File: test/toolkit.js [hunk 4]
```diff
@@ -124,3 +124,81 @@ test('h.file() outside default confine', async () => {
...
+        method : 'GET',
+        url    : '/dirtest/'
+    });
+    assertStrictEq(response.status, 200);
+    assertStrictEq(response.headers.get('content-type'), 'text/html; charset=utf-8');
+    assertStringContains(response.body, '<html>');
+    assertStringContains(response.body, 'names.json');
+});
+
+test('h.directory() forbids listing by default', async () => {
```
Comment (sholladay) [inline test/toolkit.js:184]: Can you move this test so it is before the `listing: true` tests? (Better to have default cases before edge cases.)

Comment (venikman) [reply test/toolkit.js:184]: sure
